File: deserialize.py
#!/usr/bin/python

import sys
import os
import base64
import json

def usage():
  print("usage: ./deserialize.py filename")
  print("")
  print("example: ")
  print("    $ ./serialize.py nike.jpg | ./deserialize.py test.jpg")
  print("")
  sys.exit(1)

if len(sys.argv) == 1:
  usage()

filename = sys.argv[1]

json_str = sys.stdin.read()
d = json.loads(json_str)
img_data_scheme = d["image"]

if img_data_scheme.index(",") >= 0:
  img_base64 = img_data_scheme.split(",")[1]
else:
  img_base64 = img_data_scheme.split(":")[1]
  
img = base64.b64decode(img_base64)

with open(filename, 'wb') as f:
  f.write(img)

File: deserialize.rb
#!/usr/bin/ruby

require 'json'
require 'base64'
require 'pp'

def usage
 puts "usage : "
 puts "    $ ./serialize.rb nike.jpg | #{$0} test.jpg"
 exit(0)
end

usage if ARGV.size == 0 
filename = ARGV[0]

json_str = $stdin.read

h = JSON.parse(json_str)
img_data_scheme = h["image"]

if img_data_scheme.index(",")
  img_base64 = img_data_scheme.split(",")[1]
else
  img_base64 = img_data_scheme.split(":")[1]
end

img = Base64.decode64(img_base64)

open(filename, "wb").write(img)

